protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+from the span batch. On a failed check, due to Steady Batch Derivation, the singular batch is
+immediately derived as a deposit-only block, and the remaining span batch is invalidated.
+
+There are still checks that naturally apply to the whole span batch due to its data structure:
+- A span batch only has one parent reference, since all internal batch references are implicit.
+Therefore, a failed parent check invalidates the whole span batch.
+- A failed sequencing window check will always apply to the oldest batches in a span batch, so a
+failed check of the first batch will invalidate the full span batch.
+- If `span_end.timestamp < next_timestamp` (as defined in the Span Batch specs), the whole batch is
+dropped, as it doesn't contain any new batches (this would also happen if applying timestamp checks
+to each derived singular batch individually).
+
+Those checks are done once per span batch before the batch queue derives any singular batches.
+
+# Strict Batch Ordering
+
+Strict batch ordering requires batches within and across channels to be strictly ordered.
+This removes the necessity to buffer more than one channel, and the channel bank effectively holds
+at most one staging channel, for which it collects frames, that must arrive in order.
+
+The following will explore the open design space, presenting options, and propose a solution.
+The principle I would apply to answer open design questions is that the new rules should generally
+choose the design that leads to faster progess of derivation, even if it means deriving
+empty batches earlier.
+Spoiler: Option 1 will always be the proposed option.
+
+## Out of order frames
+
+There's an open design space around how to handle some scenarios for missing or out of order frames:
+- How do we handle incoming _first_ frames while the current channel isn't closed yet?
+  - Option 1: discard the current channel and open a new one for the given frame.
+  - Option 2: discard the foreign frame and wait for the duration of the channel timeout for the
+    next correct frame and only then discard the current staging channel and open a new staging
+    channel for the next first frame.
+- How do we handle the current staging channel if an out of order frame (that is not the first
+frame) arrives? There's only one valid option, to drop the out-of-order frame, keeping the currently
+staged channel. Note that this will give the staging channel a chance to be continued or closed
+properly by the next in-order frame. If this doesn't happen, at the lastest the next first frame
+will drop the staging channel and start a new one.
+
+### Proposed solution
+
+Applied to above open question, this means choosing option 1. This has the
+additional advantage that the sync start algorithm simplifies significantly, because a fist frame
+_always_ marks the start of a new channel, and guarantees that there won't be any buffered channels
+in the channel bank.
```

Comment:
💯 this has also been part of the decision for option 1. It would otherwise be too risky to accidentally leave a channel open and cause a long chain halt.